users/test.py

```diff
@@ -36,36 +40,139 @@ def setUp(self):
         """Seed mock DB for testing"""
         self.mock_collection = mongomock.MongoClient().db.collection
         mock_data = [
-            {"username": AUTHORIZED_USER,
+            {"user_id": AUTHORIZED_USER,
              "name": AUTHORIZED_USER,
              "is_organizer": True},
-            {"username": NON_AUTHORIZED_USER,
+            {"user_id": NON_AUTHORIZED_USER,
              "name": NON_AUTHORIZED_USER,
              "is_organizer": False},
-            {"username": MALFORMATTED_IN_DB_USER}
+            {"user_id": MALFORMATTED_IN_DB_USER}
         ]
         self.mock_collection.insert_many(mock_data)
 
     def test_authorized_user_is_authorized(self):
         """An authorized user should receive authorized privileges."""
-        self.assertTrue(find_authorization_in_db(
+        self.assertTrue(app.find_authorization_in_db(
             AUTHORIZED_USER, self.mock_collection))
 
     def test_non_authorized_user_not_authorized(self):
         """A non-authorized user should not receive authorized privileges."""
-        self.assertFalse(find_authorization_in_db(
+        self.assertFalse(app.find_authorization_in_db(
             NON_AUTHORIZED_USER, self.mock_collection))
 
     def test_malformatted_user_none_authorization(self):
         """An incorrect db schema should not receive authorized privileges."""
-        self.assertFalse(find_authorization_in_db(
+        self.assertFalse(app.find_authorization_in_db(
             MISSING_USER, self.mock_collection))
 
     def test_unkown_user_not_authorized(self):
         """A user not in the db should not receive authorized privileges."""
-        self.assertFalse(find_authorization_in_db(
+        self.assertFalse(app.find_authorization_in_db(
             MALFORMATTED_IN_DB_USER, self.mock_collection))
 
 
+class TestUserUpsertion(unittest.TestCase):
+    """Test updating/inserting users into the db."""
+
+    def test_insert_valid_user(self):
+        """A valid user object should be upserted and retrieved correctly."""
+        self.mock_collection = mongomock.MongoClient().db.collection
+        user_to_insert = {
+            "user_id": USER_ID,
+            "name": USER_NAME
+        }
+        app.upsert_user_in_db(user_to_insert, self.mock_collection)
+        found_user = self.mock_collection.find_one({"user_id": USER_ID})
```

Comment:
You should check the return value of `app.upsert_user_in_db` and make sure the `user_id` matches from the returned object.

Reply:
Done https://github.com/knative-portability/large-events/pull/8/commits/e88cb3cbf6afc45fc448a0d0eb76915db152da8d